kubera/dmaas/credentials/minio.go

```diff
@@ -0,0 +1,184 @@
+	url := conn.HostName + "v3/groups"
+	body := fmt.Sprintf("token=%s;authProvider=localAuthConfig", token.JWT)
+	resp, err := client.R().
+		SetHeader("Accept", "application/json").
+		SetHeader("Cookie", body).
+		SetError(&kErr).
+		Get(url)
+
+	if err != nil || kErr.Message != "" || !resp.IsSuccess() {
+		return "", &auth.APIError{
+			Err:        err,
+			Message:    kErr.Message,
+			StatusCode: resp.StatusCode(),
+			Status:     resp.Status(),
+		}
+	}
+	// jsonpath denotes the path of backup provider id by using ProjectAccount as filter.
+	jsonPath := fmt.Sprintf("data.#(name==%s).id", "ProjectAccount")
+	// fetch group id
+	result := gjson.GetBytes(resp.Body(), jsonPath)
+
+	return result.String(), nil
+}
+
+// ID of the project in above group
+func getProjectID(token auth.Token) (string, error) {
+	client := resty.New()
+	conn := connection.Get()
+	url := conn.HostName + "v3/groups"
+	kErr := auth.KuberaError{}
+	body := fmt.Sprintf("token=%s;authProvider=localAuthConfig", token.JWT)
+	resp, err := client.R().
+		SetHeader("Accept", "application/json").
+		SetHeader("Cookie", body).
+		SetError(&kErr).
+		Get(url)
+
+	if err != nil || kErr.Message != "" || !resp.IsSuccess() {
+		return "", &auth.APIError{
+			Err:        err,
+			Message:    kErr.Message,
+			StatusCode: resp.StatusCode(),
+			Status:     resp.Status(),
+		}
+	}
+	// jsonpath denotes the path of backup provider id by using ProjectAccount as filter.
+	jsonPath := fmt.Sprintf("data.#(name==%s).projectId", "ProjectAccount")
+	// fetch project id
+	result := gjson.GetBytes(resp.Body(), jsonPath)
+
+	return result.String(), nil
+}
+
+/*
+Fetch particular backupProvider based on cloud provider name.
+Director supports a fixed set of backup providers.
+Backup providers are those that provide object storage to backup the data.
+Supported backup provider are:
+- aws
+- gcp
+- minio
+- cloudian
+*/
+func getBackupProviderID(token auth.Token, cloudProvider string) (string, error) {
+	client := resty.New()
+	conn := connection.Get()
+	url := conn.HostName + "v3/backupproviders"
+	kErr := auth.KuberaError{}
+	body := fmt.Sprintf("token=%s;authProvider=localAuthConfig", token.JWT)
+	resp, err := client.R().
+		SetHeader("Accept", "application/json").
+		SetHeader("Cookie", body).
+		SetError(&kErr).
+		Get(url)
+	if err != nil || kErr.Message != "" || !resp.IsSuccess() {
+		return "", &auth.APIError{
+			Err:        err,
+			Message:    kErr.Message,
+			StatusCode: resp.StatusCode(),
+			Status:     resp.Status(),
+		}
+	}
+	// jsonpath denotes the path of backup provider id by using name as filter.
+	jsonPath := fmt.Sprintf("data.#(name==%s).id", cloudProvider)
+	// fetch id of backup provider
+	result := gjson.GetBytes(resp.Body(), jsonPath)
+
+	return result.String(), nil
+}
+
+/*
+Add cloud credential which will be used by Kubera login to cloud provider for backup and restore.
+*/
+func addMinioCloudCredential(token auth.Token, groupID string, projectID string,
```

Comment:
The main disadvantage of using functions instead of structures is highlighted here.
See the number of arguments.